### ISSUE TITLE
Cconnection shutdown connection should not go back to pool

### ISSUE THREAD

**Issue by cincuranet** [FirebirdSQL/NETProvider#1097]
When connection is closed from server-side (`connection shutdown` response, from i.e. `DELETE FROM MON$ATTACHMENTS WHERE MON$ATTACHMENT_ID<>CURRENT_CONNECTION`), the socket is no longer usable, yet the code puts the connection back to the pool (no IO problem happened yet). This means the connection will get back to application and application will for sure have to use retry logic to get new connection and continue doing what it was doing. Not a deal-breaker as retry logic should be there anyway, but it should be handled more pro-actively.

Taken from #1095.

**Comment by cincuranet** [comment]
cc @luronumen

**Comment by mrjohnr** [comment]
hi, any news about this issue? thanks

**Comment by cincuranet** [comment]
Nope.

**Comment by mrjohnr** [comment]
is that project abandoned ?no bugs fixed for a while, no Net 7 version, in about four moths Net 8 will be released and I'm still on Net 3.1(reached end of life/no longer supported)  :(

**Comment by cincuranet** [comment]
You're more than welcome to [select a bug and fix it](https://github.com/FirebirdSQL/NETProvider/blob/master/CONTRIBUTING.md). 

What from .NET 7 you need in provider? BTW EF7 is happening [here](https://github.com/cincuranet/FirebirdSql.Data.FirebirdClient/tree/ef7), depending on my free time.

**Comment by luronumen** [comment]
Hi @cincuranet,

Any news about this issue?

Thanks in advanced!

**Comment by cincuranet** [comment]
Not really. I'm busy with other projects. Also I don't think this is critical. Retry will eventually solve it, it's just wasteful.

**Comment by luronumen** [comment]
> Not really. I'm busy with other projects. Also I don't think this is critical. Retry will eventually solve it, it's just wasteful.

Hi @cincuranet 

Thank you very much for your reply!
Yes, it is possible to work around this issue by implementing a retry mechanism.
